lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -49,6 +55,7 @@
 )
 
 AZURE_CREDENTIAL_ENV_MAPPING = dict(
+    auth_source='AZURE_AUTH_SOURCE',
```

Comment:
We actually probably don't want this in the CREDENTIAL_ENV_MAPPING, since that'd be kinda chicken-or-egg... Probably want to check it explicitly if the module arg isn't set, and not support it in the credfile at all.

Reply:
Might also want to rename `ANSIBLE_AZURE_AUTH_SOURCE` - we really should've `ANSIBLE_` namespaced all those from the start. :( Just want to make sure we don't collide with something they use.